ui/pager.go

```diff
@@ -199,6 +202,92 @@ func (m pagerModel) Update(msg tea.Msg) (pagerModel, tea.Cmd) {
 			}
 		default:
 			switch msg.String() {
+			case "e":
+				// First, determine if we can find an editor
+				editor_path, err := getEditor()
+
+				if err != nil {
+					cmd := m.showStatusMessage("No editor found!")
+					cmds = append(cmds, cmd)
+					break
+				}
+
+				// Determine if the current file is stashed or not
+				isStashed := m.currentDocument.markdownType == stashedMarkdown ||
+					m.currentDocument.markdownType == convertedMarkdown
+				
+				// Copy the contents of the document to a temporary file
+				tempFile, err := ioutil.TempFile("", "glow-edit")
+				if err != nil {
+					cmd := m.showStatusMessage("Error creating temporary file!")
+					cmds = append(cmds, cmd)
+				} else {
+					// Ensure the temporary file gets deleted
+					defer os.Remove(tempFile.Name())
+				}
+				tempFile.WriteString(m.currentDocument.Body)
+				tempFile.Close()
+
+				// Use our editor to edit the temporary file
+				cmd := exec.Command(editor_path, tempFile.Name())
+				cmd.Stdin = os.Stdin
+				cmd.Stdout = os.Stdout
+				cmd.Stderr = os.Stderr
+				err = cmd.Start()
+				if err != nil {
+					log.Println(err)
+				}
+				err = cmd.Wait()
+				if err != nil {
+					log.Println(err)
```

Comment:
There's a package-level variable `debug bool` which, when true, indicates that logs will go to a file. That variable should be checked prior to logging, otherwise logging to stdout will either mess with the UI or simply not be shown. To log to a file you can set `GLOW_LOGFILE`, by the way.

Beyond that, however, we should be properly handling errors in the UI rather than logging them.

Reply:
Ah yeah, I've seen that. Should've realized that would be ideal to use :P

I think both are good--simple status message displayed to the user and a more detailed error written to a log. I had it displaying lots of status messages; for the life of me idk why I chose not to display one for that possible error.

But I actually decided to clean up the code (I think it's a little more cleaned up, anyways) by chopping it up into several different functions. As a result, I think the displaying of status messages is a lot cleaner. If you [prefer this](https://github.com/charmbracelet/glow/compare/master...shitchell:custom) over what I have here, I'll close this PR and open one for the other branch.